### PR TITLE
[55] update content of country exportsh and its deployment for turnkey curator portal

### PR DESCRIPTION
**Pull request by stanislaw-zakrzewski**
Solves: #55 

Changes:
* Update country exporter for day0cases and query by countryISO3
* Update creation of containers for map aggregate and export
* Remove old covid related aggregation data
* Update full export to make it not dependant on individual countries
* Update github workflows for new export and aggregate container building
* Update the tests for both aggregation and export

### CODE REVIEW

File: data-serving/scripts/export-data/full_export.sh [hunk 1]
```diff
@@ -4,39 +4,29 @@
...
+require_env "${CONN:-}" "Specify MongoDB connection string in CONN"
 
-# https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-retries.html#cli-usage-retries-configure
-export AWS_MAX_ATTEMPTS=10
 
-COUNTRY_EXPORT_BUCKET="s3://${COUNTRY_EXPORT_BUCKET}"
-FULL_EXPORT_BUCKET="s3://${FULL_EXPORT_BUCKET}"
+SCRATCH="$(mktemp -d)"
+BUCKETS="${SCRATCH}/buckets.json"
+ALL_DATA="${SCRATCH}/all_data.json"
```
Comment (abhidg) [inline data-serving/scripts/export-data/full_export.sh:15]: Should be CSV as line 22 is exporting in CSV format.

File: data-serving/scripts/export-data/fields.txt [hunk 2]
```diff
@@ -11,7 +11,26 @@ demographics.ethnicity
...
+events.dateDischargeHospital
+events.intensiveCare
+events.dateAdmissionICU
+events.dateDischargeICU
+events.homeMonitoring
+events.isolated
+events.dateIsolation
+events.outcome
+events.dateDeath
+events.dateRecovered
```
Comment (abhidg) [inline data-serving/scripts/export-data/fields.txt:33]: Can't comment at line 54 - SGTF was only for the COVID-19 outbreak, can be removed.

File: data-serving/scripts/export-data/transform.py [hunk 3]
```diff
@@ -281,7 +253,7 @@ def convert_age(row: dict[str, Any], buckets: list[dict[str, Any]]) -> Optional[
 def convert_row(row: dict[str, Any], buckets: list[dict[str, Any]]) -> Optional[dict[str, Any]]:
     "Converts row for export using age buckets"
 
-    if "ObjectId" not in row["_id"]:
+    if not row["_id"].isdigit():
```
Comment (abhidg) [inline data-serving/scripts/export-data/transform.py:256]: Not a blocker for this, but we should re-visit ID as integers - can make it difficult to integrate external cases into the databases that also need to be updated, e.g. for mpox, we used `N` and `E` prefixes to integrate two different line lists into one.

Comment (stanislaw-zakrzewski) [reply data-serving/scripts/export-data/transform.py:256]: That makes sense, I'll create an issue for this.

File: data-serving/scripts/export-data/country_export.sh [hunk 4]
```diff
@@ -12,10 +12,10 @@ BUCKETS="${SCRATCH}/buckets.json"
 trap 'rm -rf "$SCRATCH"' EXIT  # Cleanup before exit
 
 FORMAT="${FORMAT:-csv,tsv,json}"
-QUERY="{\"list\": true, \"location.country\": \"$COUNTRY\"}"
+QUERY="{\"location.countryISO3\": \"$COUNTRY\"}"
```
Comment (jim-sheldon) [inline data-serving/scripts/export-data/country_export.sh:15]: data should be approved by a senior curator?

File: data-serving/scripts/export-data/transform.py (hunk 3 above)
Comment (jim-sheldon) [reply data-serving/scripts/export-data/transform.py:256]: I think we should discuss this further. I would tend toward using a separate curation-specific field, as this seems to be a curation-specific action we would want to keep an audit trail for.

File: data-serving/scripts/export-data/country_export.sh (hunk 4 above)
Comment (stanislaw-zakrzewski) [reply data-serving/scripts/export-data/country_export.sh:15]: I completely forgot about this, good call, thanks.

File: data-serving/scripts/export-data/full_export.sh
[the remark this answers is not in the document]
Comment (stanislaw-zakrzewski) [reply data-serving/scripts/export-data/full_export.sh:1]: I'll add an issue for that

File: data-serving/scripts/export-data/country_export.sh (hunk 4 above)
Comment (jim-sheldon) [reply data-serving/scripts/export-data/country_export.sh]: To be resolved in another PR?

**Review by jim-sheldon — APPROVED** [review]

**Pull request merged**